### PR TITLE
Call consensus reads

### PR DESCRIPTION
**Pull request by FelixMoelder**
This PR adds a rule for calling rbt consensus reads.

Currently, only overlapping reads are going to be merged, while duplicates remain separated.

While the current behavior is supposed for panel data we could optionally add an additional rule annotation reads duplicate groups.
In case that rule would be activated consensus read calling would also merge duplicates into a single read.
Which might be useful to improve read quality instead of dropping duplicates.

### CODE REVIEW

**Comment by FelixMoelder** [comment]
Reminder for myself:
Environments for rbt and varlociraptor needs to be updated when new releases are available.
These include handling of strand information

**Comment by FelixMoelder** [comment]
Todo: Template for `rbt vcf-report` needs to be updated to working with the latest rbt version.

**Comment by FelixMoelder** [comment]
The custom report template has been removed.
Instead all modifications of the default template have been moved in a separate file by @fxwiegand.
Moving modifications also allows using the latest version of the default template including several bug fixes and improvement.
Future changes to the `rbt vcf-report` template will be directly included when every the rbt dependency gets updated.

File: .test/config-chm-eval/config.yaml [hunk 1]
```diff
@@ -119,6 +119,12 @@ params:
     MarkDuplicates: "VALIDATION_STRINGENCY=lenient"
   gatk:
     BaseRecalibrator: ""
+  varlociraptor_preprocess:
+    max_depth: "200"
```
Comment (johanneskoester) [inline .test/config-chm-eval/config.yaml:123]:
```suggestion
    max_depth: 200
```

File: .test/config-no-candidate-filtering/config.yaml [hunk 2]
```diff
@@ -77,6 +77,12 @@ params:
   gatk:
     BaseRecalibrator: ""
     applyBQSR: ""
+  varlociraptor_preprocess:
+    max_depth: "200"
```
Comment (johanneskoester) [inline .test/config-no-candidate-filtering/config.yaml:81]:
```suggestion
    max_depth: 200
```

File: .test/config-sra/config.yaml [hunk 3]
```diff
@@ -70,6 +70,12 @@ params:
   gatk:
     BaseRecalibrator: ""
     applyBQSR: ""
+  varlociraptor_preprocess:
+    max_depth: "200"
```
Comment (johanneskoester) [inline .test/config-sra/config.yaml:74]:
```suggestion
    max_depth: 200
```

File: .test/config/config.yaml [hunk 4]
```diff
@@ -78,6 +78,12 @@ params:
   gatk:
     BaseRecalibrator: ""
     applyBQSR: ""
+  varlociraptor_preprocess:
+    max_depth: "200"
```
Comment (johanneskoester) [inline .test/config/config.yaml:82]:
```suggestion
    max_depth: 200
```

File: .test/config_primers/config.yaml [hunk 5]
```diff
@@ -74,6 +74,12 @@ params:
   gatk:
     BaseRecalibrator: ""
     applyBQSR: ""
+  varlociraptor_preprocess:
+    max_depth: "200"
```
Comment (johanneskoester) [inline .test/config_primers/config.yaml:78]:
```suggestion
    max_depth: 200
```

File: config/config.yaml [hunk 6]
```diff
@@ -128,3 +132,9 @@ params:
   gatk:
     BaseRecalibrator: ""
     applyBQSR: ""
+  varlociraptor_preprocess:
+    max_depth: "200" # Set when processing panel data with high coverage
```
Comment (johanneskoester) [inline config/config.yaml:136]:
```suggestion
    max_depth: 200 # Set to desired depth when processing panel data with high coverage
```

File: workflow/envs/varlociraptor.yaml [hunk 7]
```diff
@@ -2,4 +2,4 @@ channels:
   - conda-forge
   - bioconda
 dependencies:
-  - varlociraptor =2.5.0
+  - varlociraptor =2.6.0
```
Comment (johanneskoester) [inline workflow/envs/varlociraptor.yaml:5]:
```suggestion
  - varlociraptor =2.6.1
```

File: workflow/rules/annotation.smk [hunk 8]
```diff
@@ -48,7 +48,7 @@ rule annotate_vcfs:
     log:
         "logs/annotate-vcfs/{prefix}.log"
     params:
-        extra="-Xmx4g",
+        extra="-Xmx4g", #Never used
```
Comment (johanneskoester) [inline workflow/rules/annotation.smk:51]:
```suggestion
```

File: workflow/rules/calling.smk [hunk 9]
```diff
@@ -18,20 +18,20 @@ rule varlociraptor_preprocess:
...
         candidates=get_candidate_calls(),
-        bam="results/recal/{sample}.sorted.bam",
-        bai="results/recal/{sample}.sorted.bai"
+        bam=lambda w: get_group_bams(w),
+        bai=lambda w: get_group_bams(w, bai=True),
     output:
         "results/observations/{group}/{sample}.{caller}.{scatteritem}.bcf"
     params:
-        omit_isize = "--omit-insert-size" if is_activated("primers/trimming") else ""
+        max_depth="--max-depth {}".format(config["params"]["varlociraptor_preprocess"].get("max_depth", "200"))
```
Comment (johanneskoester) [inline workflow/rules/calling.smk:26]:
```suggestion
        max_depth="--max-depth {}".format(config["params"]["varlociraptor_preprocess"].get("max_depth", 200))
```

File: workflow/rules/report.smk [hunk 10]
```diff
@@ -1,4 +1,5 @@
 rule vcf_report:
+    threads: 8
```
Comment (johanneskoester) [inline workflow/rules/report.smk:2]: Move threads directly above shell. Not enforced by snakefmt yet, but this is more consistent.

File: workflow/schemas/config.schema.yaml [hunk 11]
```diff
@@ -168,6 +168,12 @@ properties:
       activate:
         type: boolean
 
+  calc_consensus_reads:
```
Comment (johanneskoester) [inline workflow/schemas/config.schema.yaml:171]: Also add the max_depth setting in case it is still missing.

**Review by johanneskoester — CHANGES_REQUESTED** [review]

**Comment by FelixMoelder** [comment]
Suggested changes have been collectively added in a single commit.

**Review by johanneskoester — APPROVED** [review]

**Pull request merged**